src/Framework/Analyzer/src/DelegateEndpoints/DetectMismatchedParameterOptionality.Fixer.cs

```diff
@@ -0,0 +1,38 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using System.Collections.Generic;
+using Microsoft.CodeAnalysis.CSharp;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using Microsoft.CodeAnalysis.CodeFixes;
+using Microsoft.CodeAnalysis.Operations;
+using Microsoft.CodeAnalysis.Editing;
+
+namespace Microsoft.AspNetCore.Analyzers.DelegateEndpoints;
+
+public partial class DelegateEndpointFixer : CodeFixProvider
```

Comment:
I was following the pattern we currently use for analyzers because I think it makes sense. One main file where all the definitions are registered and then each codefix in its own file.

Reply:
It's far less likely we would re-use a single codefix for all of these scenarios. The analyzers were useful to share because they were detecting the same thing - viz calls to MapGet. It's less common to want to do that with the codefix. In addition, code fixes aren't on a hot path